src/value.rs

```diff
@@ -14,36 +14,37 @@ use std::fmt;
 use std::hash::Hash;
 use std::sync::Arc;
 
-/// Value of some type.
-///
-/// The _unit value_ is the only value of the _unit type_.
-/// This is the basis for everything we are doing.
-/// Because there is only a single unit value, there is no information contained in it.
-/// Instead, we wrap unit values in sum and product values to encode information.
-///
-/// A _sum value_ wraps another value.
-/// The _left sum value_ `L(a)` wraps a value `a` from the _left type_ `A`.
-/// The _right sum value_ `R(b)` wraps a value `b` from the _right type_ `B`.
-/// The type of the sum value is the _sum type_ `A + B` of the left type and the right type.
-///
-/// We represent the false bit as a left value that wraps a unit value.
-/// The true bit is represented as a right value that wraps a unit value.
-///
-/// A _product value_ `(a, b)` wraps two values:
-/// a value `a` from the _left type_ `A` and a value `b` from the _right type_ `B`.
-/// The type of the product value is the _product type_ `A × B` of the left type and the right type.
-///
-/// We represent bit strings (tuples of bits) as trees of nested product values
-/// that have bit values (sum values wrapping the unit value) at their leaves.
+/// A Simplicity value.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub enum Value {
-    /// Unit value
+    /// Unit value.
+    ///
+    /// The unit value is the only value of the unit type `1`.
+    /// Because there is only one value to choose from,
+    /// this means that the unit value doesn't contain any information.
+    ///
+    /// The unit value is wrapped in left and right values
+    /// to encode information.
     Unit,
-    /// Sum value that wraps a left value
+    /// Left value that wraps an inner value.
+    ///
+    /// A left value is the first option for a sum type `A + B`.
+    /// It means we go left and wrap a value of type `A`.
+    /// The choice of going left encodes one bit.
     Left(Arc<Value>),
-    /// Sum value that wraps a right value
+    /// Right value that wraps an inner value.
+    ///
+    /// A right value is the second option for a sum type `A + B`.
+    /// It means that we go right and wrap a value of type `B`.
+    /// The choice of going right encodes one bit.
     Right(Arc<Value>),
-    /// Product value that wraps a left and a right value
+    /// Product value that wraps a left and a right value.
+    ///
+    /// A product value is the only option for a product type `A × B`.
+    /// It means we go both left and right and
+    /// that we wrap values of both type `A` and type `B`.
+    /// There is no choice,
+    /// so the information is contained solely in the wrapped values.
```

Comment:
In 7459af049bca4c548626a56e350dc074755d5725:

This "only option" business is definitely confusing. The number of values in a product type is equal to the product of the numbers of values of the factor types. I would drop this paragraph and say "Since there is no ambiguity about which types are occupied, no additional bits are used and the information is contained solely in the wrapped values.". If you want to say this at all.